platforms/software/dependency-management/src/main/java/org/gradle/internal/component/ResolutionFailureHandler.java

```diff
@@ -216,6 +225,19 @@ private <FAILURE extends ResolutionFailure> AbstractResolutionFailureException d
             .filter(describer -> describer.canDescribeFailure(failure))
             .findFirst()
             .map(describer -> describer.describeFailure(failure, schema))
+            .map(this::reportExceptionAsProblem)
             .orElseThrow(() -> new IllegalStateException("No describer found for failure: " + failure)); // TODO: a default describer at the end of the list that catches everything instead?
     }
+
+    private AbstractResolutionFailureException reportExceptionAsProblem(AbstractResolutionFailureException exception) {
+        Problem problem = problemsService.getInternalReporter().create(builder -> {
+            builder.id(TextUtil.screamingSnakeToKebabCase(exception.getFailure().getProblemId().name()), "variant resolution error", GradleCoreProblemGroup.variantResolution())
```

Comment:
I think the resolution failure display name should be more specific, because it's intended to be rendered on a UI. `ResolutionFailureProblemId` should have a `displayName()` method.

Reply:
Good idea, I'll add that.  I have to rebase this change on top of a different update first.